src/app/src/test/kotlin/org/vaccineimpact/orderlyweb/tests/unit_tests/templates/rules/FreemarkerTestRule.kt

```diff
@@ -0,0 +1,107 @@
+package org.vaccineimpact.orderlyweb.tests.unit_tests.templates.rules
+
+import com.gargoylesoftware.htmlunit.html.HtmlPage
+import com.gargoylesoftware.htmlunit.html.HTMLParser
+import freemarker.template.Configuration
+import freemarker.template.Template
+import freemarker.template.TemplateExceptionHandler
+import freemarker.template.Version
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+import java.io.File
+import java.io.StringWriter
+import java.io.Writer
+import java.lang.IllegalStateException
+import java.net.URL
+import java.util.*
+import javax.xml.transform.Source
+import org.xmlmatchers.transform.XmlConverters.the
+import com.gargoylesoftware.htmlunit.StringWebResponse
+import com.gargoylesoftware.htmlunit.WebClient
+
+
+//This is a simplified kotlinised version of this: https://github.com/Todderz/freemarker-unit-test
+
+class FreemarkerTestRule(val templateName: String, val templatePath: String = "templates",
+                         val includeTemplates: List<String> = listOf()) : TestRule
+{
+    companion object
+    {
+        const val anyUrl = "http://localhost"
+        val anyWindow = WebClient().currentWindow
+    }
+
+    private var template: Template? = null
+
+    override fun apply(base: Statement, description: Description): Statement
+    {
+        return object : Statement() {
+            override fun evaluate() {
+
+                loadTemplate()
+                //execute the test
+                base.evaluate()
+            }
+        }
+    }
+
+    private fun getTemplate() : Template
+    {
+        return template ?: throw IllegalStateException()
+    }
+
+    private fun loadTemplate()
+    {
+        val config = configureTemplateLoader()
+        template = config.getTemplate(templateName)
+    }
+
+    private fun configureTemplateLoader(): Configuration
+    {
+        val config = Configuration()
+        config.setDirectoryForTemplateLoading(File(templatePath))
+        config.incompatibleImprovements = Version(2, 3, 20)
+        config.defaultEncoding = "UTF-8"
+        config.locale = Locale.UK
+        config.templateExceptionHandler = TemplateExceptionHandler.RETHROW_HANDLER
+
+        //Assume we should include standard layout file
+        config.addAutoInclude("layouts/layout.ftl")
```

Comment:
`layouts/layoutwide.ftl` is also auto-included here: https://github.com/vimc/orderly-web/blob/master/src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/app_start/SparkApp.kt#L39

In fact, can we share the config between app and tests somehow to ensure we're testing with the correct config?

Reply:
It should be doing that anyway actually, with the default templatePath, I think I just thought it was going to be more sensitive when it didn't automatically pull in the layoutwide template (which of course it wouldn't as it's in a different directory)

Reply:
This is what I meant: https://github.com/vimc/orderly-web/pull/50